### ISSUE TITLE
ED2: No way to set custom PFT numbers

### ISSUE THREAD

**Issue by ashiklom** [PecanProject/pecan#2257]
`write.config.xml.ed2` currently has this block:

https://github.com/PecanProject/pecan/blob/develop/models/ed/R/write.configs.ed.R#L489-L498

Based on that, it seems that there is no way to define the mapping between BETY PFTs and ED PFT numbers (1-17) without modifying the `pftmapping` that ships with the `PEcAn.ED2` package. This seems to me like a glaring design flaw, since it effectively makes it impossible to run ED with any PFTs other than the ones defined in the package.

The _ideal_ solution is for the PFT mapping to live somewhere in BETY (perhaps in our fancy new `attributes` table?). But, in the meantime, I propose that we give each PFT an optional `<ed2_pft_number>` XML tag that forces ED to use that PFT number. For instance:

```xml
<pft>
  <name>umbs.early_hardwood</pft>
  <ed2_pft_number>9</ed2_pft_number>
</pft>
```

If that tag is missing, we fall back on the existing `pftmapping`. This has the advantage of being backwards compatible (because `ed2_pft_number` is a new, optional tag missing from all prior XMLs), and that it should be straightforward to add code that sets it via the database later. It also has the advantage of being unambiguously ED-specific (which we want, because this is an ED-specific problem).

Assuming nobody has any issues with this, I'll open a PR shortly. I'm happy to can consider alternative solutions, but I need this to work now, and this seems like a quick, straightforward, and sensible fix.

(@serbinsh @istfer @mdietze -- tagging you because you all may some thoughts on this).

**Comment by serbinsh** [comment]
Agreed, the attributes table could be a place for this!  I know we have been discussing this design flaw for many years now and @kooper mentioned several times in the past the need to store the parameter defaults for ED2 pfts on bety rather than a file shipped with the package.  That said one way to run with a custom pft is to overwrite all parameter values, correct?  That is select one of the defaults but then create a bety pft that contains all of the Ed params and provides new values.  Or a combination of that with setting pft <constants>. Not ideal but a work around.  

That said, we should strive to eliminate pftmapping.csv

**Comment by ashiklom** [comment]
This was closed by #2259. Formal conversations about this should be moved to the more general issue #2371.
